kube-derive/src/lib.rs

```diff
@@ -161,6 +161,37 @@ mod resource;
 /// ## `#[kube(served = true)]`
 /// Sets the `served` property to `true` or `false`.
 ///
+/// ## `#[kube(deprecated [= "warning"])]`
+/// Sets the `deprecated` property to `true`.
+///
+/// ```
+/// # use serde::{Serialize, Deserialize};
+/// # use kube::core::{Resource, CustomResourceExt};
+/// # use kube_derive::CustomResource;
+/// # use schemars::JsonSchema;
+///
+/// #[derive(CustomResource, Clone, Debug, Deserialize, Serialize, JsonSchema)]
+/// #[kube(group = "clux.dev", version = "v1", kind = "Foo", namespaced, deprecated)]
+/// struct FooSpec {
+///     info: String,
+/// }
+/// ```
+///
+/// Optionally, the `deprecationWarning` property can be set using the following example.
+///
+/// ```
+/// # use serde::{Serialize, Deserialize};
+/// # use kube::core::{Resource, CustomResourceExt};
+/// # use kube_derive::CustomResource;
+/// # use schemars::JsonSchema;
+///
+/// #[derive(CustomResource, Clone, Debug, Deserialize, Serialize, JsonSchema)]
+/// #[kube(group = "clux.dev", version = "v1", kind = "Foo", namespaced, deprecated = "my warning")]
+/// struct FooSpec {
+///     info: String,
+/// }
```

Comment:
good docs, but maybe we don't need a full example here for conciseness. maybe it's enough to simply display the line?:

```rust
#[kube(deprecated = "Replaced by other CRD")]
```

Reply:
Oh yeah sure thing; I will update it.

Reply:
Updated in fde63554d8845776aafa9039e71106393a19c9d2